react/components/ContainerStateSelector.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, useState, useEffect } from 'react'
+import { Button } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { path } from 'ramda'
+
+import SelectSkuMessage from './SelectSkuMessage'
+import StoreSelectedQuery from './StoreSelectedQuery'
+import AddressWithList from './AddressWithList'
+
+interface Props {
+  googleMapsKey?: string
+  showSelectSkuMessage: boolean
+}
+
+function usePickupFromSession() {
+  const [favoritePickup, setFavoritePickup] = useState<FavoritePickup | undefined>(undefined)
+
+  useEffect(() => {
+    let isCurrent = true
+    const sessionPromise = (window as any).__RENDER_8_SESSION__.sessionPromise
```

Comment:
🤔 this should perhaps come from something like `vtex.render-runtime`, but it's ok for the time being

Reply:
I agree